app/__init__.py

```diff
@@ -78,135 +77,340 @@ def create_app():
 
     @app.cli.command()
     @click.option('--filename', default=None)
-    def import_dois_from_csv(filename):
-        """Import CSV file with DOI's.
+    def import_from_csv(filename):
+        """Import csv file with doi's and OJS url's.
+
+        Imports the DOI's and OJS url's from a csv file into the database.
+        For development purposes there is a file with 100 entries you can use.
+        With `--filename` you can pass the filename to the function.
 
         Args:
             filename: filename with path.
         """
         import pandas as pd
         from app.models import Doi
-
-        default_csv_filename = 'app/static/data/PKP_20171220_100.csv'
-
-        if not filename:
-            filename = default_csv_filename
-
-        df = pd.read_csv(filename, encoding='utf8', parse_dates=True)
-        df = df.drop_duplicates(subset='doi')
-        num_rows = len(df.index)
-        num_added = 0
-        num_already_in = 0
-        for index, row in df.iloc[:num_rows].iterrows():
-            doi = row['doi']
-            result = Doi.query.filter_by(doi=doi).first()
-            if result is None:
-                doi = Doi(
-                    doi=doi,
-                    source_type='file',
-                    source_file=filename,
-                    source_json=None
-                )
-                db.session.add(doi)
-                num_added += 1
-            else:
-                num_already_in += 1
-        db.session.commit()
-        print(num_added, 'doi\'s added to database.')
-        print(num_already_in, 'doi\'s already in database.')
+        from app.models import Import
+        from app.models import Url
+        import re
+
+        try:
+            df = pd.read_csv(filename, encoding='utf8', parse_dates=True)
+            df = df.drop_duplicates(subset='doi')
+            imp = Import('<file '+filename+'>', df.to_string())
+            db.session.add(imp)
+            db.session.commit()
+        except:
```

Comment:
Could we add a more detailed error output here (and eventually in other places)? I was trying to load a file and initially it wasn't possible to discern db errors from wrong filenames etc

Reply:
Also, I was wondering if there needs to be a `return` or other flow-control statement at the end of the `except` because otherwise it goes on even though no file has been loaded

Reply:
`flask db upgrade` is the command to be used. should work with the new commit (added the env.py file).

Reply:
regarding error output and pipeline workflow: will do that, when the basic pipeline runs from beginning to end.